### PR TITLE
collect performance stats when OCS operation fails for scale and perfomance test cases

### PR DESCRIPTION
**Pull request by Akarsha-rai**
Collect performance stats and saves them in file in json format.
Performance stats include:
- IOPs and throughput percentage of cluster
- CPU, memory consumption of each nodes

Signed-off-by: Akarsha <akrai@redhat.com>

### CODE REVIEW

**Comment by Akarsha-rai** [comment]
Collects performance stats on path

$ cat /tmp/failed_testcase_ocs_logs_1583304684/performance_stats/performance
{"iops_percentage": 0.005555555555555556, "throughput_percentage": 4.687497599999999e-06, "master_node_utilization": {"control-plane-0": {"cpu": 39, "memory": 20}, "control-plane-1": {"cpu": 37, "memory": 19}, "control-plane-2": {"cpu": 42, "memory": 19}}, "worker_node_utilization": {"compute-0": {"cpu": 10, "memory": 20}, "compute-1": {"cpu": 7, "memory": 14}, "compute-2": {"cpu": 6, "memory": 14}}}


Fixes: https://github.com/red-hat-storage/ocs-ci/issues/1537

File: tests/helpers.py [hunk 1]
```diff
@@ -2056,3 +2057,46 @@ def modify_osd_replica_count(resource_name, replica_count):
...
+    performance_stats = {}
+
+    # Get iops and throughput percentage of cluster
+    iops_percentage = ceph_obj.get_iops_percentage()
+    throughput_percentage = ceph_obj.get_throughput_percentage()
+
+    # ToDo: Get iops and throughput percentage of each nodes
+
+    # Get the cpu and memory of each nodes
+    master_node_utilization = node.get_node_resource_utilization(node_type='master')
```
Comment (ramkiperiy) [inline tests/helpers.py]: In PR https://github.com/red-hat-storage/ocs-ci/pull/1586 I have changed this function name, make sure to update this.

File: tests/helpers.py [hunk 2]
```diff
@@ -2066,3 +2067,52 @@ def modify_osd_replica_count(resource_name, replica_count):
...
+    throughput_percentage = ceph_obj.get_throughput_percentage()
+
+    # ToDo: Get iops and throughput percentage of each nodes
+
+    # Get the cpu and memory of each nodes from adm top
+    master_node_utilization_from_adm_top = node.get_node_resource_utilization_from_adm_top(node_type='master')
+    worker_node_utilization_from_adm_top = node.get_node_resource_utilization_from_adm_top(node_type='worker')
+
+    # Get the cpu and memory from describe of nodes
+    master_node_utilization_from_oc_describe = node.get_node_resource_utilization_from_oc_describe(node_type='master')
```
Comment (Shrivaibavi) [inline tests/helpers.py]: Please split lines

File: tests/helpers.py (hunk 1 above)
Comment (Akarsha-rai) [reply tests/helpers.py]: done

File: tests/helpers.py [hunk 3]
```diff
@@ -2066,3 +2067,52 @@ def modify_osd_replica_count(resource_name, replica_count):
...
+    worker_node_utilization_from_oc_describe = node.get_node_resource_utilization_from_oc_describe(node_type='worker')
```
[the remark this answers is not in the document]
Comment (Akarsha-rai) [reply tests/helpers.py]: done

File: tests/helpers.py (hunk 2 above)
Comment (Akarsha-rai) [reply tests/helpers.py]: done

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 4]
```diff
@@ -382,3 +382,12 @@ def pytest_runtest_makereport(item, call):
             call.start,
             call.stop
         )
+
+    # Get the performance metrics when tests fails for scale or performance tag
+    from tests.helpers import collect_performance_stats
```
Comment (SvetlanaRedHat) [inline ocs_ci/framework/pytest_customization/ocscilib.py:387]: Cant you have imports on top of the code?

Comment (Akarsha-rai) [reply ocs_ci/framework/pytest_customization/ocscilib.py:387]: Was hitting cyclic import error. So called here

**Pull request merged**